packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,1556 @@
+      throw new Error('EBS volumes are required when using instance types other than r3 or i3.');
+    }
+
+    // Fine-grained access control requires node-to-node encryption, encryption at rest,
+    // and enforced HTTPS.
+    if (advancedSecurityEnabled) {
+      if (!nodeToNodeEncryptionEnabled) {
+        throw new Error('Node-to-node encryption is required when fine-grained access control is enabled.');
+      }
+      if (!encryptionAtRestEnabled) {
+        throw new Error('Encryption-at-rest is required when fine-grained access control is enabled.');
+      }
+      if (!enforceHttps) {
+        throw new Error('Enforce HTTPS is required when fine-grained access control is enabled.');
+      }
+    }
+
+    let cfnVpcOptions: CfnDomain.VPCOptionsProperty | undefined;
+    if (props.vpcOptions) {
+      cfnVpcOptions = {
+        securityGroupIds: props.vpcOptions.securityGroups.map((sg) => sg.securityGroupId),
+        subnetIds: props.vpcOptions.subnets.map((subnet) => subnet.subnetId),
+      };
+    }
+
+    // Setup logging
+    const logGroups: logs.ILogGroup[] = [];
+
+    if (props.logging?.slowSearchLogEnabled) {
+      this.slowSearchLogGroup = props.logging.slowSearchLogGroup ??
+        new logs.LogGroup(scope, 'SlowSearchLogs', {
+          retention: logs.RetentionDays.ONE_MONTH,
+        });
+
+      logGroups.push(this.slowSearchLogGroup);
+    };
+
+    if (props.logging?.slowIndexLogEnabled) {
+      this.slowIndexLogGroup = props.logging.slowIndexLogGroup ??
+        new logs.LogGroup(scope, 'SlowIndexLogs', {
+          retention: logs.RetentionDays.ONE_MONTH,
+        });
+
+      logGroups.push(this.slowIndexLogGroup);
+    };
+
+    if (props.logging?.appLogEnabled) {
+      this.appLogGroup = props.logging.appLogGroup ??
+        new logs.LogGroup(scope, 'AppLogs', {
+          retention: logs.RetentionDays.ONE_MONTH,
+        });
+
+      logGroups.push(this.appLogGroup);
+    };
+
+    let logGroupResourcePolicy: LogGroupResourcePolicy | null = null;
+    if (logGroups.length > 0) {
+      const logPolicyStatement = new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ['logs:PutLogEvents', 'logs:CreateLogStream'],
+        resources: logGroups.map((lg) => lg.logGroupArn),
+        principals: [new iam.ServicePrincipal('es.amazonaws.com')],
+      });
+
+      // Use a custom resource to set the log group resource policy since it is not supported by CDK and cfn.
+      // https://github.com/aws/aws-cdk/issues/5343
+      logGroupResourcePolicy = new LogGroupResourcePolicy(this, 'ESLogGroupPolicy', {
+        policyName: 'ESLogPolicy',
+        policyStatements: [logPolicyStatement],
+      });
+    }
+
+    // Create the domain
+    this.domain = new CfnDomain(this, 'Resource', {
+      domainName: this.physicalName,
+      accessPolicies: unsignedBasicAuthEnabled
+        ? (props.accessPolicies ?? []).concat(unsignedAccessPolicy)
```

Comment:
You treat `accessPolicies` here as an array of `iam.PolicyStatement`. It should actually be an `iam.PolicyDocument` JSON dictionary.

Reply:
Fixed.